projects/plugins/jetpack/class.jetpack.php

```diff
@@ -4020,12 +4019,12 @@ public function admin_page_load() {
 					wp_safe_redirect( self::admin_url( 'page=jetpack' ) );
 					exit;
 				case 'unlink':
-					$redirect = isset( $_GET['redirect'] ) ? $_GET['redirect'] : '';
+					$redirect = isset( $_GET['redirect'] ) ? $_GET['redirect'] : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
 					check_admin_referer( 'jetpack-unlink' );
 					self::log( 'unlink' );
 					$this->connection_manager->disconnect_user();
 					self::state( 'message', 'unlinked' );
-					if ( 'sub-unlink' == $redirect ) {
+					if ( 'sub-unlink' === $redirect ) {
 						wp_safe_redirect( admin_url() );
 					} else {
 						wp_safe_redirect( self::admin_url( array( 'page' => $redirect ) ) );
```

Comment:
Yay footguns. Unless you use `sanitize_key()` above (which only allows lowercase letters, numbers, underscore and hyphen, no idea if that's valid here), this needs a `rawurlencode()`.
```suggestion
						wp_safe_redirect( self::admin_url( array( 'page' => rawurlencode( $redirect ) ) ) );
```